telegram_menu/navigation.py

```diff
@@ -456,9 +459,9 @@ async def capture_user_input(self, label: str, context: Optional[CallbackContext
         """Process the user input in the last message updated."""
         last_menu_message = self._menu_queue[-1]
         if self._message_queue:
-            last_app_message = self._message_queue[-1]
-            if last_app_message.time_alive > last_menu_message.time_alive:
-                last_menu_message = last_app_message
+            for last_app_message in self._message_queue[::-1]:
```

Comment:
Why adding a for loop here ? I would expect that the user input is used by the latest message only, not by an older one upper in the queue

Reply:
If a user clicks an inline button in a message that is not the last one I expect `text_input` is redirected to the message that contains the inline button I clicked, not to the last sent message. Do you agree?

Reply:
Yes that should work. Would be good to add a unit-test to validate this use-case though